SwEnt-Group8/Street-work-app
# Feat/Update and set gamification in relevant parts of the UI

## Overview
The objective of this Pull Request is to implement the gamification feature in the relevant part of the app. To do so, a `Snackbar` is displayed whenever the user wins points, with the amount of points obtained. This `Snackbar` allows the user to go see directly the progression screen updated with the new score by clicking on the "see Progression" part of the `Snackbar.`
An increase of score and a display of the `Snackbar` is added in three parts of the app in this Pull Request: when the user creates an event, when he joins an event and when he adds a new friend.

## Changes
- A new function `updateAndDisplayPoints` is created and used in order to increase the score of the user and display the `Snackbar`
- Creating an event, joining an event and adding a friend now adds points
- Details of the Progression UI screen are also updated (add missing icon for medals and update achievements description)

## How to test
1. Go to the Map Screen
2. Select any park, and add or join an event

You should see a `Snackbar` appear, with the option to "See Progression" (see screenshot if it is not clear), you can click on the "See Progression" part, which will redirect you to the Progression Screen.

3. Check that your score increased


## Screenshot
<img width="237" alt="image" src="https://github.com/user-attachments/assets/28140b54-8fb0-4845-b494-6643c559a658">
